clippy_lints/src/formatting.rs

```diff
@@ -133,6 +157,47 @@ fn check_assign(cx: &EarlyContext<'_>, expr: &Expr) {
     }
 }
 
+/// Implementation of the `SUSPICIOUS_UNARY_OP_FORMATTING` lint.
+fn check_unop(cx: &EarlyContext<'_>, expr: &Expr) {
+    if let ExprKind::Binary(ref binop, ref lhs, ref rhs) = expr.kind {
+        if !differing_macro_contexts(lhs.span, rhs.span) && !lhs.span.from_expansion() {
+            // span between BinOp LHS and RHS
+            let binop_span = lhs.span.between(rhs.span);
+            // if RHS is a UnOp
+            if let ExprKind::Unary(op, ref un_rhs) = rhs.kind {
+                // from UnOp operator to UnOp operand
+                let unop_operand_span = rhs.span.until(un_rhs.span);
+                if let (Some(binop_snippet), Some(unop_operand_snippet)) =
+                    (snippet_opt(cx, binop_span), snippet_opt(cx, unop_operand_span))
+                {
+                    let binop_str = BinOpKind::to_string(&binop.node);
+                    // no space after BinOp operator and space after UnOp operator
+                    if binop_snippet.ends_with(binop_str) && unop_operand_snippet.ends_with(' ') {
```

Comment:
You can put all this in one `if_chain!`:
```rust
if_chain! {
    if let ExprKind::Binary(ref binop, ref lhs, ref rhs) = expr.kind;
    if !differing_macro_contexts(lhs.span, rhs.span) && !lhs.span.from_expansion();
    // span between BinOp LHS and RHS
    let binop_span = lhs.span.between(rhs.span);
    // if RHS is a UnOp
    if let ExprKind::Unary(op, ref un_rhs) = rhs.kind;
    // from UnOp operator to UnOp operand
    let unop_operand_span = rhs.span.until(un_rhs.span);
    if let Some(binop_snippet) = snippet_opt(cx, binop_span);
    if let Some(unop_operand_snippet) = snippet_opt(cx, unop_operand_span);
    let binop_str = BinOpKind::to_string(&binop.node);
    // no space after BinOp operator and space after UnOp operator
    if binop_snippet.ends_with(binop_str) && unop_operand_snippet.ends_with(' ');
    then {
        ...
    }
}
```

After this, this should be ready to merge.

Reply:
Done! Very satisfying to use macro. :)
Thanks for all your feedback!